vchernyshoff/homework_5/main.py

```diff
@@ -0,0 +1,177 @@
+    "
+
+    result = strip_comments("apples, pears # and bananas\ngrapes\nbananas !apples", ["#", "!"])
+    # "apples, pears\ngrapes\nbananas"
+    """
+    new_strings = ''
+    strings = input_string.split('\n')
+    for string in strings:
+        new_string = ''
+        for next_symbol in string:
+            if next_symbol in markers:
+                new_string = string.split(next_symbol)[0]
+        if new_string == '':
+            new_string = string
+        new_strings += new_string.strip() + '\n'
+    new_strings = new_strings[:-1]  # cut last \n
+
+    print(new_strings)
+    return new_strings
+
+
+def format_duration(input_seconds):
+    """
+    Your task is to write a function which formats a duration, given as a number of seconds, in a human-friendly way.
+    The function must accept a non-negative integer. If it is zero, it just returns "now". Otherwise, the duration is expressed as a combination of years, days, hours, minutes and seconds.
+
+    Examples:
+
+    format_duration(62)    # returns "1 minute and 2 seconds"
+    format_duration(3721)  # returns "1 hour, 2 minutes and 1 second"
+
+    Assume that year is 365 days
+    """
+    def check_period(seconds, period):
+        quantity = None
+        if input_seconds // period > 1:
+            quantity = seconds // period
+            seconds -= quantity * period
+            quantity = str(quantity)
+        return quantity, seconds
+
+    if input_seconds == 0:
+        return 'now'
+
+
+    YEAR = 31536000
+    DAY = 86400
+    HOUR = 3600
+    MIN = 60
+    # years, days, hours, seconds = None, None, None, None
+
+    # result = datetime.timedelta(seconds=input_seconds)
+    # print(result.days, result.minutes, result.seconds)
+    seconds = input_seconds
+
+    years, seconds = check_period(seconds, YEAR)
+    days, seconds = check_period(seconds, DAY)
+    hours, seconds = check_period(seconds, HOUR)
+    minutes, seconds = check_period(seconds, MIN)
+
+    result = ''
+
+    if years:
+        # print(years, 'years,', end=' ')
+        result += years + ' years, '
+    if days:
+        # print(days, 'days,', end=' ')
+        result += days + ' days, '
+    if hours:
+        # print(hours, 'hours,', end=' ')
+        result += hours + ' hours, '
+    if minutes:
+        # print(minutes, 'minutes', end=' and ')
+        result += minutes + ' minutes and '
+    # print(seconds, 'seconds')
+    result += str(seconds) + ' seconds'
+
+    print(result)
+    return result
+
+
+
+def is_isogram(input_string):
+    """
+    An isogram is a word that has no repeating letters, consecutive or non-consecutive.
+    Implement a function that determines whether a string that contains only letters is an isogram.
+    Assume the empty string is an isogram. Ignore letter case.
+
+    is_isogram("Dermatoglyphics" ) == true
+    is_isogram("aba" ) == false
+    is_isogram("moOse" ) == false # -- ignore letter case
+    """
+    all_symbols = list(str(input_string).lower())
```

Comment:
Input_string it's str and you again do it str)

Reply:
If user will enter digits, like '12341' - it will exception.
This function make available use digits and check it.
12345 - True
123145 - False

Reply:
are you sure that you want to accept it here? It makes that argument implicit (because of it name input_**string**). Just add type hints, like `input_string: str`

Reply:
No problem ) I deleted str() function